applications/MeshingApplication/custom_processes/metrics_hessian_process.cpp

```diff
@@ -470,29 +489,27 @@ void ComputeHessianSolMetricProcess::InitializeVariables(Parameters ThisParamete
     // Get default variables
     Parameters default_parameters = GetDefaultParameters();
 
-    // Set variables
-    mMinSize = ThisParameters["minimal_size"].GetDouble();
-    mMaxSize = ThisParameters["maximal_size"].GetDouble();
-    mEnforceCurrent = ThisParameters["enforce_current"].GetBool();
-
     // In case we have isotropic remeshing (default values)
-    if (ThisParameters["anisotropy_remeshing"].GetBool() == false) {
-        mEstimateInterpError = default_parameters["hessian_strategy_parameters"]["estimate_interpolation_error"].GetBool();
-        mInterpError = default_parameters["hessian_strategy_parameters"]["interpolation_error"].GetDouble();
-        mMeshConstant = default_parameters["hessian_strategy_parameters"]["mesh_dependent_constant"].GetDouble();
-        mRatioReferenceVariable = default_parameters["anisotropy_parameters"]["reference_variable_name"].GetString();
-        mAnisotropicRatio = default_parameters["anisotropy_parameters"]["hmin_over_hmax_anisotropic_ratio"].GetDouble();
-        mBoundLayer = default_parameters["anisotropy_parameters"]["boundary_layer_max_distance"].GetDouble();
-        mInterpolation = ConvertInter(default_parameters["anisotropy_parameters"]["interpolation"].GetString());
-    } else {
-        mEstimateInterpError = ThisParameters["hessian_strategy_parameters"]["estimate_interpolation_error"].GetBool();
-        mInterpError = ThisParameters["hessian_strategy_parameters"]["interpolation_error"].GetDouble();
-        mMeshConstant = ThisParameters["hessian_strategy_parameters"]["mesh_dependent_constant"].GetDouble();
-        mRatioReferenceVariable = ThisParameters["anisotropy_parameters"]["reference_variable_name"].GetString();
-        mAnisotropicRatio = ThisParameters["anisotropy_parameters"]["hmin_over_hmax_anisotropic_ratio"].GetDouble();
-        mBoundLayer = ThisParameters["anisotropy_parameters"]["boundary_layer_max_distance"].GetDouble();
-        mInterpolation = ConvertInter(ThisParameters["anisotropy_parameters"]["interpolation"].GetString());
-    }
+    const bool default_values = !ThisParameters["anisotropy_remeshing"].GetBool();
+    const Parameters considered_parameters = default_values ? default_parameters : ThisParameters;
+    mThisParameters.AddValue("minimal_size", ThisParameters["minimal_size"]);
+    mThisParameters.AddValue("maximal_size", ThisParameters["maximal_size"]);
+    mThisParameters.AddValue("enforce_current", ThisParameters["enforce_current"]);
+    mThisParameters.AddValue("anisotropy_remeshing", ThisParameters["anisotropy_remeshing"]);
+    mThisParameters.AddValue("enforce_anisotropy_relative_variable", ThisParameters["enforce_anisotropy_relative_variable"]);
+    mThisParameters.AddValue("estimate_interpolation_error", considered_parameters["hessian_strategy_parameters"]["estimate_interpolation_error"]);
+    mThisParameters.AddValue("interpolation_error", considered_parameters["hessian_strategy_parameters"]["interpolation_error"]);
```

Comment:
Right now, if `anisotropy_remeshing` is set to false, the interpolation error is taken as the default.

I would avoid this since even if you are using an isotropic remeshing, you might want to set a different interpolation error. 
```suggestion
    mThisParameters.AddValue("interpolation_error", ThisParameters["hessian_strategy_parameters"]["interpolation_error"]);
```